File: TTCS_CQ3_FE/fe-englishstudy-web/src/utils/services/collectionService.js
import { apiRequest } from '../apiClient';


export async function fetchCollections() {
  return apiRequest('/api/collections', { method: 'GET', auth: true });
}


export async function createCollection(name) {
  return apiRequest('/api/collections', {
    method: 'POST',
    auth: true,
    body: { collectionName: name }
  });
}


export async function deleteCollection(collectionId) {
  return apiRequest(`/api/collections/${collectionId}`, {
    method: 'DELETE',
    auth: true
  });
}


export async function fetchCollectionVocabs(collectionId) {
  return apiRequest(`/api/collections/${collectionId}/vocabs`, {
    method: 'GET',
    auth: true
  });
}


export async function addVocabToCollection(collectionId, vocabId) {
  return apiRequest(`/api/collections/${collectionId}/vocabs/${vocabId}`, {
    method: 'POST',
    auth: true
  });
}

export async function updateCollectionName(collectionId, newName) {
  return apiRequest(`/api/collections/${collectionId}`, {
    method: 'PUT',
    auth: true,
    body: { name: newName }
  });
}


export async function removeVocabFromCollection(collectionId, vocabId) {
  return apiRequest(`/api/collections/${collectionId}/vocabs/${vocabId}`, {
    method: 'DELETE',
    auth: true
  });
}

File: TTCS_CQ3_BE/src/main/java/com/example/demo/service/CollectionService.java
package com.example.demo.service;

import com.example.demo.dto.response.CollectionVocabResponse;
import com.example.demo.dto.response.CollectionResponseDTO;

import java.util.List;

public interface CollectionService {

    List<CollectionResponseDTO> getUserCollections(Long userId);

    CollectionResponseDTO createCollection(Long userId, String collectionName);

    void deleteCollection(Long userId, Long collectionId);

    void addVocabToCollection(Long userId, Long collectionId, Long vocabId);

    List<CollectionVocabResponse> getVocabsInCollection(Long userId, Long collectionId);

    CollectionResponseDTO updateCollectionName(Long userId, Long collectionId, String newName);

    void removeVocabFromCollection(Long userId, Long collectionId, Long vocabId);
}
